OpenDreamServer/Dream/Objects/MetaObjects/DreamMetaObjectWorld.cs

```diff
@@ -18,6 +18,15 @@ class DreamMetaObjectWorld : DreamMetaObjectRoot {
             if (fps.Value != null) {
                 dreamObject.SetVariable("tick_lag", new DreamValue(10.0f / fps.GetValueAsInteger()));
             }
+            
+            //system_type value should match the defines in Defines.dm
+            if (Environment.OSVersion.Platform is PlatformID.Unix or PlatformID.MacOSX or PlatformID.Other) {
```

Comment:
Setting the variable when creating the world object will end up leaving `system_type` writable instead of constant. Moving this to `OnVariableGet` would cause this to always return the same value, even after attempting to set it.